cpp/src/Slice/Parser.cpp

```diff
@@ -660,64 +658,55 @@ namespace
                 }
             } while (pos != string::npos);
+        }
 
-            const string link = "{@link";
-            pos = 0;
-            do
+        // Fix any link tags using the provided link formatter.
+        const string link = "{@link ";
+        pos = comment.find(link, pos);
+        while (pos != string::npos)
+        {
+            string::size_type endpos = comment.find('}', pos);
+            if (endpos != string::npos)
             {
-                pos = comment.find(link, pos);
-                if (pos != string::npos)
+                // Extract the linked to identifier.
+                string::size_type identStart = comment.find_first_not_of(" \t", pos + link.size());
+                string::size_type identEnd = comment.find_last_not_of(" \t", endpos) + 1;
+                string ident = comment.substr(identStart, identEnd - identStart);
+
+                // Then erase the entire '{@link foo}' tag from the comment.
+                comment.erase(pos, endpos - pos + 1);
+
+                // Split the link into 'class' and 'member' components (links are of the form 'class#member').
+                string memberComponent;
+                string::size_type hashPos = ident.find('#');
+                if (hashPos != string::npos)
                 {
-                    comment.erase(pos, link.size() + 1); // Erase trailing white space too.
-                    string::size_type endpos = comment.find('}', pos);
-                    if (endpos != string::npos)
-                    {
-                        string ident = comment.substr(pos, endpos - pos);
-                        comment.erase(pos, endpos - pos + 1);
-
-                        //
-                        // Replace links of the form {@link Type#member} with "Type.member".
-                        //
-                        string::size_type hash = ident.find('#');
-                        string rest;
-                        if (hash != string::npos)
-                        {
-                            rest = ident.substr(hash + 1);
-                            ident = ident.substr(0, hash);
-                            if (!ident.empty())
-                            {
-                                if (!rest.empty())
-                                {
-                                    ident += "." + rest;
-                                }
-                            }
-                            else if (!rest.empty())
-                            {
-                                ident = rest;
-                            }
-                        }
-
-                        comment.insert(pos, ident);
-                    }
+                    memberComponent = ident.substr(hashPos + 1);
+                    ident.erase(hashPos);
                 }
-            } while (pos != string::npos);
+                else
+                {
+                    memberComponent = "";
+                }
```

Comment:
Seems this else is not required, `memberComponent` is already `""` in this case.

Reply:
Removed!